pkg/operator/apiserver/controller/workload/workload.go

```diff
@@ -356,6 +424,55 @@ func (c *Controller) updateOperatorStatus(ctx context.Context, previousStatus *o
 	return nil
 }
 
+func (c *Controller) deleteWorkload(ctx context.Context, workloadName string) (err error) {
+	deploymentAvailableCondition := applyoperatorv1.OperatorCondition().
+		WithType(fmt.Sprintf("%sDeployment%s", c.conditionsPrefix, operatorv1.OperatorStatusTypeAvailable))
+
+	workloadDegradedCondition := applyoperatorv1.OperatorCondition().
+		WithType(fmt.Sprintf("%sWorkloadDegraded", c.conditionsPrefix))
+
+	status := applyoperatorv1.OperatorStatus()
+	defer func() {
+		status = status.WithConditions(deploymentAvailableCondition, workloadDegradedCondition)
+		if applyError := c.operatorClient.ApplyOperatorStatus(ctx, c.controllerInstanceName, status); applyError != nil {
+			err = applyError
+		}
+	}()
+
+	if _, getErr := c.deploymentLister.Deployments(c.targetNamespace).Get(workloadName); getErr != nil && !apierrors.IsNotFound(getErr) {
+		deploymentAvailableCondition = deploymentAvailableCondition.
+			WithStatus(operatorv1.ConditionFalse).
+			WithReason("DeletionError")
+
+		workloadDegradedCondition = workloadDegradedCondition.
+			WithStatus(operatorv1.ConditionTrue).
+			WithReason("DeletionError").
+			WithMessage(getErr.Error())
+		return getErr
+
+	} else if getErr == nil {
+		if deleteErr := c.kubeClient.AppsV1().Deployments(c.targetNamespace).Delete(ctx, workloadName, metav1.DeleteOptions{}); deleteErr != nil && !apierrors.IsNotFound(deleteErr) {
+			deploymentAvailableCondition = deploymentAvailableCondition.
+				WithStatus(operatorv1.ConditionFalse).
+				WithReason("DeletionError")
+
+			workloadDegradedCondition = workloadDegradedCondition.
+				WithStatus(operatorv1.ConditionTrue).
+				WithReason("DeletionError").
+				WithMessage(deleteErr.Error())
+			return deleteErr
+		}
+	}
+
+	deploymentAvailableCondition = deploymentAvailableCondition.
+		WithStatus(operatorv1.ConditionTrue).
+		WithReason("AsExpected")
+	workloadDegradedCondition = workloadDegradedCondition.
+		WithStatus(operatorv1.ConditionFalse)
```

Comment:
What is the value of preserving these conditions (`Available=True`, `Degraded=False`) if no workload exists for them?

Or are we interested in measuring the availability of OIDC provider? Is it possible? And even then this probably isn't the right place to indicate it, right?